.semaphore/semaphore.yml

```diff
@@ -449,6 +453,43 @@ blocks:
       commands:
       - make ci
 
+- name: 'networking-calico'
+  run:
+    when: "change_in(['/networking-calico/'])"
+  task:
+    prologue:
+      commands:
+      - sudo apt-get install -y python-all-dev python3-all-dev python3-pip
+      - sudo pip3 install tox
+    jobs:
+      - name: 'Unit and FV tests (tox)'
+        commands:
+          - checkout
+          - cd networking-calico
+          - tox
+      # TODO: Needs reinstating.
```

Comment:
Is there an issue with running this part of the CI?

Reply:
@neiljerram I just hadn't gotten around to it and didn't have time to figure it out just yet, but didn't want to lose it!

It seems to check out a specific branch of networking-calico named "devstack-test" which I suspect doesn't exist in monorepo anymore, so I stopped there because it was more than I had time to investigate :)

Reply:
It should be fine.  "devstack-test" is a scratch branch that the CI creates itself.

Reply:
doh, you're right. I totally misread that.

Reply:
@neiljerram do you happen to know _why_ we need to check out a new branch for this?

Reply:
Not sure.  It might be because we need to guarantee that
```
ncref=`git rev-parse --abbrev-ref HEAD`
```
will return a short local branch name, which is in turn because we need to tell the devstack process to reclone the repo again, via
```
enable_plugin networking-calico $ncdir $ncref
```
here: https://github.com/projectcalico/calico/blob/master/networking-calico/devstack/bootstrap.sh#L116

Could be that with just a Semaphore checkout, we don't necessarily get a nice "master" or "release-v3.21", because of Semaphore having merged the PR branch with current master/release branch tip?